profiles/prefix/darwin/macos/features/clang-lld/make.defaults

```diff
@@ -5,3 +5,8 @@ LD=ld64.lld
 
 # Several packages fail to build without this
 lt_cv_apple_cc_single_mod=yes
+
+# This is the sort of equivalent of -Wl,--as-needed
+# It's already set in profiles/prefix/darwin, but profiles/features/llvm
+# appends --as-needed to it, so here we set it back to the previous value
+LDFLAGS="-Wl,-dead_strip_dylibs"
```

Comment:
just to be sure, ld.lld doesn't support --as-needed on macOS targets?  (e.g. it behaves as ld64?)

Reply:
That's why it's called ld64.lld and not ld.lld

Reply:
scratch that, I see you override LD to ld64.lld.  So then it makes more sense to redefine LDFLAGS too.  I'd group them in your commit so it make sense.

This is ok, it's duplicating a bit, but not to much, and I think it's much more clean than the split-profile approach.  Thanks!

Reply:
Done